### ISSUE TITLE
Priority of EXIF tag vs Kmatrix

### ISSUE THREAD

**Issue by hcjghr** [openMVG/openMVG#554]
Hi,

while looking at the _main_SfMInit_ImageListing.cpp_ I have noticed that the EXIF tag has priority over manually input intrinsic matrix (Kmatrix). Is there any specific reason for this, because from my point of view it makes more sense that the intrinsic matrix should be used if provided (as it can be assumed the user knows for a fact the parameters) no matter if the image has a valid EXIF data.

thanks

**Comment by pmoulon** [comment]
Thanks @hcjghr for having noting this.
I have designed the manual intrinsic matrix option for images that does not have EXIF data.
So your right, EXIF tag have more priority.

@ORNis, @rperrot @mitjap  What is your point on this?
Does the -k option ```[-k|--intrinsics] Kmatrix: "f;0;ppx;0;f;ppy;0;0;1"``` must have the priority over all other initialization?

regarding that point, I think it should be cool to have also a pixel focal length initialization from FocalPlaneXResolution/FocalPlaneYResolution/FocalPlaneResolutionUnit in order to initialize some camera intrinsics that are not registered in the sensor database, but that provide information in the EXIF.
See here: http://photo.stackexchange.com/a/40870

**Comment by rperrot** [comment]
Maybe a force option would be interesting ? 

if force is here, it replace everything that is specified manualy. If not, automatic or already present data is kept, but if data is unavailable it uses the provided one. 

If you don't want to specify a force option, my opinion is to use the provided info. So -k should replace the EXIF data. If not, this is missleading because when you provide something you expect that it's used, not "use it if info not available" (or in that case option should be renamed to something like -default-intrinsics).

**Comment by mitjap** [comment]
I would agree with @rperrot. I recommend flags `--default-intrinsics` and `-f|--force`, while `force` option is only valid if `default` exists.

Flag `-k|--intrinsics` could then be short version for `--default-intrinsics --force`.

**Comment by ORNis** [comment]
I agree with @rperrot and @hcjghr too, user provided data should have priority over exif+database generated data.

Maybe two other suggestions:
- As suggested by @pmoulon, we could maybe use other existing exif tags to provide a "true" (i.e. without database) automatic initialization (I think that Foc35 tag is a good one too)
- I heard that maybe a calibration module is planned for v.1.1. So we could also discuss of a centralized-community way to share pre computed calibrations (a kind of web service?) in another issue.

**Comment by hcjghr** [comment]
Hi,

I created a really simple PR where I just change the order of priority between user provided (either -k or -f and exif tag) as I saw @robert128 had same issue. I'm not exactly sure that I understand what you meant by --default-intrinsics and -f|--force. Isn't it the same as saying -k is "forced" if valid and otherwise use exif tag? Or am I missing something?

The PR is at [https://github.com/hcjghr/openMVG/tree/develop_init_order](https://github.com/hcjghr/openMVG/tree/develop_init_order)
